src/index.cpp

```diff
@@ -3098,7 +3235,6 @@ template <typename T, typename TagT, typename LabelT> int Index<T, TagT, LabelT>
     _delete_set->insert(location);
     _location_to_tag.erase(location);
     _tag_to_location.erase(tag);
-
     return 0;
 }
 
```

Comment:
How are handling deletes and related clean up?

Reply:
With current strategy, we don't need to do much after delete. We have made frozen_points as medoids, so deleting medoids is not possible. Also when we do compact_data after deleting and repositioning frozen_points, medoids are also repositioned accordingly as in filtered streaming scenario medoids = frozen_points.